pulsar-websocket/src/main/java/com/yahoo/pulsar/websocket/AbstractWebSocketHandler.java

```diff
@@ -81,10 +81,14 @@ public void onWebSocketConnect(Session session) {
                     log.warn("[{}] WebSocket Client [{}] is not authorized on topic {}", session.getRemoteAddress(), role,
                             topic);
                     close(WebSocketError.NotAuthorizedError);
+                } else {
+                    createClient(session);
```

Comment:
do you think it make sense to complete handshake ` super.onWebSocketConnect(session);` after successfully authenticating client, means after calling `createClient(session)`?